test/lint/lint-circular-dependencies.sh

```diff
@@ -13,6 +13,7 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "checkpoints -> validation -> checkpoints"
     "index/txindex -> validation -> index/txindex"
     "policy/fees -> txmempool -> policy/fees"
+    "policy/policy -> policy/settings -> policy/policy"
```

Comment:
Let's try not to introduce new circular dependencies. I had a quick look at #15639 and it seems that we can keep all the moved code in the `policy` translation unit and prevent the circular dependency?